Refactor of `travis.ps1` into `ci.psm1`

RDIL opened the pull request:
<!-- Anything that looks like this is a comment and can't be seen after the Pull Request is created. -->  

## PR Summary
Refactors `travis.ps1`. Important for maintainers: I removed the webhook and badge stuff as neither exist anymore. I also merged the build and test phase with the bootstrap phase.   
<!-- Summarize your PR between here and the checklist. -->  

<!-- Provide a little reasoning as to why this Pull Request helps and why you have opened it. -->

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is **very** breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [X] Issue filed: <!-- Number/link of that issue here --> #8686 is related
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

RDIL committed:
Start refactor

Commit:
the start!

Commit:
move Get-ReleaseTag

Commit:
move Get-DailyBadge

Commit:
Port Set-DailyBuildBadge

Commit:
removed commit checking

Commit:
[feature] trigger feature tests

Commit:
more

RDIL committed:
[feature] debug CI?

RDIL committed:
[feature] Remove outdated get and set methods

RDIL committed:
Update mac.yml

RDIL committed:
[feature] update tests

RDIL commented:
Oops uh I think I deleted the feature tests, I'll try to add them back

RDIL committed:
[feature] fix

RDIL commented on .vsts-ci/linux.yml:
```diff
@@ -80,23 +80,23 @@ jobs:
 
   - powershell: |
-      tools/travis.ps1 -Stage Bootstrap
+      tools/ci.psm1 -Stage Bootstrap
```
I see, I will wrap it in a function once I have access to vscode

RDIL committed:
Update linux.yml

RDIL committed:
Update mac.yml

adityapatwardhan commented:
@RDIL This is great. Thanks for taking this up. I have an open PR https://github.com/PowerShell/PowerShell/pull/8775 which also does a bunch of changes to `travis.ps1`. Once it is merged, you should rebase to take in those changes as well.

RDIL commented:
@adityapatwardhan thanks.  Also once that is merged, it would be lovely if you could help out by porting it over here.

Commit:
[feature] is this going to work

Commit:
fixthat

Commit:
[feature] run additional tests

Commit:
[feature] fix issue?

Commit:
[feature] fix name so azure devops knows where to look

Commit:
[feature] done??!

Commit:
[feature] thats better

TylerLeonhardt commented on .vsts-ci/linux.yml:
```diff
@@ -80,23 +80,24 @@ jobs:
   - powershell: |
       $ErrorActionPreference = 'continue'
-      tools/travis.ps1 -NugetKey $(NUGET_KEY)
+      tools/ci.psm1 -NugetKey $(NUGET_KEY)
```
This doesn't look right. You probably wanted to invoke a function that's _inside_ the psm1.

RDIL replied:
It isn't letting me accept the suggestion

TylerLeonhardt commented on .vsts-ci/linux.yml:
```diff
@@ -80,23 +80,24 @@ jobs:
 
   - powershell: |
-      tools/travis.ps1 -Stage Failure
+      Write-Host "Tests failed."
```
I think you could probably just log success or fail inside of the psm1 and then you can safely take out this task.

TylerLeonhardt commented on .vsts-ci/linux.yml:
```diff
@@ -80,23 +80,24 @@ jobs:
 
   - powershell: |
-      tools/travis.ps1 -Stage Success
+      Write-Host "Tests did not fail!"
```
I think you could probably just log success or fail inside of the psm1 and then you can safely take out this task.

TylerLeonhardt commented on .vsts-ci/mac.yml:
```diff
@@ -66,12 +67,12 @@ jobs:
 
   - powershell: |
-      tools/travis.ps1 -Stage Failure
+      Write-Host "Tests failed."
```
I think you could probably just log success or fail inside of the psm1 and then you can safely take out this task.

TylerLeonhardt commented on .vsts-ci/mac.yml:
```diff
@@ -66,12 +67,12 @@ jobs:
 
   - powershell: |
-      tools/travis.ps1 -Stage Success
+      Write-Host "Tests did not fail!"
```
I think you could probably just log success or fail inside of the psm1 and then you can safely take out this task.

RDIL replied:
I moved it to the PSM1.

TylerLeonhardt commented on tools/travis.ps1:
```diff
@@ -1,425 +0,0 @@
-Import-Module $PSScriptRoot/../build.psm1 -Force
-Import-Module $PSScriptRoot/packaging -Force
-
-function Send-DailyWebHook
```
Do we still need to send the DailyWebHook @TravisEz13?

TylerLeonhardt reviewed (commented):
Some comments to start with. I can take a closer look later today

Commit:
remove old daily build code

Commit:
[feature] fix broken call

Commit:
[feature] removed after-fail as suggested by @TylerLeonhardt

Commit:
again removed the task for if tests did not fail

Commit:
e

Commit:
[feature] trigger feature tests + revert!

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +594,212 @@ function Invoke-AppveyorFinish
+    return $preReleaseVersion
+}
+
+function Invoke-LinuxTests
```
@TravisEz13 here

adityapatwardhan commented:
@RDIL The PR with changes to `travis.ps1` is merged. Hence the merge conflict. Let me know if you need any help in merging it.

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

RDIL commented:
@adityapatwardhan conflict resolved.

RDIL commented on .vsts-ci/mac.yml:
```diff
@@ -55,7 +55,8 @@ jobs:
   - powershell: |
-      tools/travis.ps1 -Stage Bootstrap
+      Import-Module .\tools\ci.psm1
+      Invoke-LinuxTests
```
I added it in the PSM1 but for some reason the CI says not found instead of running tests.

RDIL committed:
Remove duplicated code

RDIL committed:
Remove bootstrap step

RDIL committed:
Remove Mac bootstrap

RDIL committed:
Fix pester/bootstrap tests

Commit:
Fix Mac

RDIL committed:
Fix nuget key

RDIL committed:
Begin to fix

RDIL committed:
Update ci.psm1

RDIL committed:
Update linux.yml

RDIL committed:
Update mac.yml

RDIL committed:
Update linux.yml

RDIL committed:
Update linux.yml

RDIL committed:
Update mac.yml

RDIL commented:
omg I actually fixed the tests 🐙

RDIL committed:
Update linux.yml

RDIL committed:
done??!

RDIL committed:
added @adityapatwardhan 's changes

RDIL committed:
[feature] test extra tests

RDIL committed:
[feature] umm okay?

RDIL committed:
update test name

RDIL committed:
[feature] done for now!!  (now need review)

RDIL commented:
/cc @JamesWTruher

RDIL committed:
remove appveyor comments so I can claim this PR mitigates #8686

RDIL committed:
[feature] cleanup

RDIL committed:
fixed Get-ReleaseTag function

RDIL committed:
[feature] more fixing of #8686

RDIL committed:
[feature] add back check

fixes CI

RDIL commented:
@TravisEz13 I fixed everything, the PR is now ready for re-review!  😄

RDIL committed:
[feature] fix some #8686 issues

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +584,176 @@ function Invoke-AppveyorFinish
+    Write-Host -Foreground Green "Executing ci.psm1 -BootStrap `$isPR='$isPr' - $commitMessage"
+    # Make sure we have all the tags
+    Sync-PSTags -AddRemoteIfMissing
+    Start-PSBootstrap -Package:$createPackages
```
Okay

RDIL replied:
What line does the bootstrap phase end on?

RDIL commented:
Wait what why is `install-ps` its own task 😕

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +584,176 @@ function Invoke-AppveyorFinish
+
+function Invoke-LinuxTests
+{
+    $createPackages = $isFullBuild
```
Okay

RDIL committed:
move it I think?

RDIL committed:
split up bootstrap

RDIL committed:
Update linux.yml

RDIL committed:
Update mac.yml

RDIL committed:
Update linux.yml

RDIL committed:
Update mac.yml

RDIL committed:
Update ci.psm1

RDIL committed:
Fix tag failures

RDIL committed:
[feature] Move and fix stuff

RDIL commented:
@TravisEz13 all feedback addressed.

TravisEz13 committed:
Update .vsts-ci/mac.yml

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +578,178 @@ function Invoke-AppveyorFinish
+
+function Invoke-LinuxTests
+{
+    Sync-PSTags -AddRemoteIfMissing
```
Okay

RDIL replied:
Fixed

TravisEz13 committed:
Update .vsts-ci/linux.yml

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL committed:
Address feedback

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +578,178 @@ function Invoke-AppveyorFinish
+{
+    Write-Host -Foreground Green "Executing ci.psm1 Bootstrap `$isPR='$isPr' - $commitMessage"
+    # Make sure we have all the tags
+    Sync-PSTags -AddRemoteIfMissing
```
Will do

RDIL replied:
Fixed.  Not gonna do that if it has a chance of lots of errors.

Commit:
Address feedback

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

Commit:
[feature] commit a thing

adityapatwardhan commented on tools/windows/Reset-PWSHSystemPath.ps1:
```diff
@@ -1,3 +1,5 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
```
Add an empty line after the license text

adityapatwardhan commented on tools/ci.psm1:
```diff
@@ -593,3 +578,178 @@ function Invoke-AppveyorFinish
+                Start-NativeExecution -sb {dotnet nuget push $package --api-key $NugetKey --source "$env:NUGET_URL/api/v2/package"} -IgnoreExitcode
+            }
+        }
+        if ($IsLinux)
```
Missing changes PR #8775 

https://github.com/PowerShell/PowerShell/blob/fed0ef0a2011e0cf65afdc94ac2b3eda4d82050c/tools/travis.ps1#L381-L397

RDIL replied:
Thanks for the feedback, I'll get on this at once.

RDIL replied:
Should be done.

adityapatwardhan commented on tools/windows/Reset-PWSHSystemPath.ps1:
```diff
@@ -1,3 +1,5 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
 <#
```
Add an empty line after license text

RDIL committed:
Update Reset-PWSHSystemPath.ps1

RDIL committed:
Merge remote-tracking branch 'upstream/master' into patch-11

Commit:
include adityapatwardhan's new changes

Commit:
[feature] github can you please stop being annoying thanks

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

Commit:
[feature] run feature tests

RDIL commented:
@adityapatwardhan @TravisEz13 please re-review whenever you have time.

RDIL commented on tools/ci.psm1:
```diff
@@ -513,6 +498,15 @@ function Invoke-AppveyorFinish
+	        {
+	            Write-Log "Package found: $package"
+	        }
+	        else
```
Seems this is failing the windows ci build

RDIL committed:
Fixed some broken indents

RDIL committed:
[feature] just warn for now to get windows passing again

RDIL committed:
[feature] rename update-appveyortestresults

RDIL committed:
[feature] fix name

RDIL committed:
[feature] fix quotes

RDIL committed:
[feature] fix?

RDIL committed:
Please work I beg

RDIL committed:
Please work

RDIL committed:
[feature] try with feature tests

RDIL commented:
I fixed it on normal tests, now just waiting to see if feature tests will pass

RDIL commented:
Passing feature tests!

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

RDIL commented:
@iSazonov do you mind taking a quick look over this?  (pretty sure everyone else isn't awake yet lol)

iSazonov commented on .vsts-ci/linux.yml:
```diff
@@ -80,24 +80,17 @@ jobs:
 
   - powershell: |
-      tools/travis.ps1 -Stage Bootstrap
+      Import-Module ./tools/ci.psm1
```
Interesting, auto-loading does not work?

RDIL replied:
Because it is now a PSM1

iSazonov commented on .vsts-ci/linux.yml:
```diff
@@ -80,24 +80,17 @@ jobs:
     condition: succeeded()
 
   - powershell: |
       $ErrorActionPreference = 'continue'
```
Does it work for module below?

RDIL replied:
It should

RDIL replied:
I moved it into the PSM1 and out of the YAMLs

iSazonov commented on .vsts-ci/mac.yml:
```diff
@@ -55,24 +55,17 @@ jobs:
     condition: succeeded()
 
   - powershell: |
       $ErrorActionPreference = 'continue'
```
The same.

iSazonov commented on tools/releaseTools.psm1:
```diff
@@ -44,7 +44,7 @@ $Script:community_login_map = @{
 
 $Script:attribution_ignore_list = @(
-    'dependabot[bot]@users.noreply.github.com'
+    'dependabot[bot]@users.noreply.github.com' # dependency bumping bot
```
I'd put the comment on separate line.

RDIL replied:
Moved it.  Thanks for the review.

RDIL committed:
move erroroptions

RDIL committed:
move 1/2

RDIL committed:
[feature] move 2/2

RDIL committed:
[feature] address feedback

RDIL committed:
[feature] rerun

RDIL commented:
CC @JamesWTruher @SteveL-MSFT

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +594,212 @@ function Invoke-AppveyorFinish
         throw $_
```
Okay

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +594,212 @@ function Invoke-AppveyorFinish
+            {
+                # Create and package Raspbian .tgz
+                Start-PSBuild -PSModuleRestore -Clean -Runtime linux-arm -Configuration 'Release'
+                Start-PSPackage @packageParams -Type tar-arm -SkipReleaseChecks
```
already fixed

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on .vsts-ci/linux.yml:
```diff
@@ -80,26 +80,19 @@ jobs:
-
   # Uploads any packages as an artifact
   - powershell: |
       Get-ChildItem -Path *.rpm, *.deb, *.tar.gz, TestPackage.zip -Recurse | Select-Object -ExpandProperty FullName | ForEach-Object {
```
It says outdated suggestions can't be applied

RDIL committed:
[feature] fix stuff?

RDIL commented:
@TravisEz13 I believe I addressed all the feedback I could in https://github.com/PowerShell/PowerShell/pull/8822/commits/564afa113667dce5d1733f020bde0ca5974d7951

RDIL commented on tools/ci.psm1:
```diff
@@ -593,3 +594,212 @@ function Invoke-AppveyorFinish
+                # Create and package Raspbian .tgz
+                Start-PSBuild -PSModuleRestore -Clean -Runtime linux-arm -Configuration 'Release'
+                Start-PSPackage @packageParams -Type tar-arm -SkipReleaseChecks
+            }
```
Not sure, I know I lost a few of the daily build checks in the move because my git likes to act up

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL committed:
fix dupes

RDIL commented on tools/ci.psm1:
```diff
@@ -402,13 +402,13 @@ function Invoke-AppVeyorAfterTest
 
         Write-Host -ForegroundColor Green 'Upload CodeCoverage artifacts'
         $codeCoverageArtifacts | ForEach-Object {
             Push-Artifact -Path $_ -Name 'CodeCoverage'
```
oops I think I put it on the wrong line

TravisEz13 committed:
remove extra line

RDIL committed:
fixed

RDIL committed:
address feedback

RDIL commented on tools/ci.psm1:
```diff
@@ -587,9 +593,244 @@ function Invoke-AppveyorFinish
+                {
+                    Write-Error -Message "Package NOT found: $package"
+                }
+                Write-Log "pushing $package to $env:NUGET_URL"
```
fixed

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -587,9 +593,244 @@ function Invoke-AppveyorFinish
+            {
+                Write-Error -Message "Package NOT found: $package"
+            }
+            # Publish the packages to the nuget feed if:
```
fixed

RDIL commented on tools/ci.psm1:
```diff
@@ -513,6 +510,15 @@ function Invoke-AppveyorFinish
 
         $artifacts = New-Object System.Collections.ArrayList
         foreach ($package in $packages) {
+            if (Test-Path $package)
```
sure!

RDIL committed:
[feature] fixed indents

TravisEz13 committed:
some code was duplicated so I cleaned it up

TravisEz13 committed:
delete extra line

RDIL committed:
fix

RDIL committed:
fixed

TravisEz13 committed:
fixed misc issues locally

RDIL commented:
I saw, thanks!

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

TravisEz13 committed:
remove duplicate line that is in the wrong place

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

iSazonov commented on tools/ci.psm1:
```diff
@@ -790,9 +793,9 @@ function Invoke-LinuxTests
             Start-PSBuild -PSModuleRestore -Clean -Runtime linux-arm -Configuration 'Release'
             $armPackage = Start-PSPackage @packageParams -Type tar-arm -SkipReleaseChecks
-	    Copy-Item $armPackage -Destination "${env:BUILD_ARTIFACTSTAGINGDIRECTORY}" -Force
+	        Copy-Item $armPackage -Destination "${env:BUILD_ARTIFACTSTAGINGDIRECTORY}" -Force
```
Indentation.

RDIL committed:
fixed indent

iSazonov commented on tools/ci.psm1:
```diff
@@ -792,7 +792,7 @@ function Invoke-LinuxTests
             Start-PSBuild -PSModuleRestore -Clean -Runtime linux-arm -Configuration 'Release'
             $armPackage = Start-PSPackage @packageParams -Type tar-arm -SkipReleaseChecks
-	        Copy-Item $armPackage -Destination "${env:BUILD_ARTIFACTSTAGINGDIRECTORY}" -Force
+	    Copy-Item $armPackage -Destination "${env:BUILD_ARTIFACTSTAGINGDIRECTORY}" -Force
```
Again. :-) 
(can miss this enchanted string? :-) )

RDIL replied:
In vscode it is indented properly, I think it’s just github

iSazonov replied:
Perhaps a invisible control symbol. Please remove all start spaces and retype.

RDIL replied:
Will do.

RDIL replied:
Does it look fixed now @iSazonov ?  I tried deleting the line and reindenting it.

iSazonov replied:
Yes, now looks good!

RDIL replied:
Great, thanks for the review :)

RDIL committed:
Fix it??

iSazonov commented:
LGTM.

RDIL commented:
@adityapatwardhan please update your review.

adityapatwardhan commented on tools/ci.psm1:
```diff
@@ -225,18 +225,19 @@ function Invoke-AppVeyorInstall
-        $appveyorRemoteCredential = [PSCredential]::new("$env:COMPUTERNAME\$userName", $ss)
-	    $appveyorRemoteCredential | Export-Clixml -Path "$env:TEMP\AppVeyorRemoteCred.xml" -Force
+        $ciRemoteCredential = [PSCredential]::new("$env:COMPUTERNAME\$userName", $ss)
+	$ciRemoteCredential | Export-Clixml -Path "$env:TEMP\AppVeyorRemoteCred.xml" -Force
```
Indentation does not look right.

RDIL replied:
Will resolve when I get back home.  Thanks for noticing.

RDIL replied:
Thanks.  I believe it is fixed now :)

adityapatwardhan reviewed (approved):
LGTM with one minor comment

RDIL committed:
Remove ghost Unicode odd intending stuff

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

Commit:
replace write-host with write-verbose & cleanup

Commit:
Merge remote-tracking branch 'upstream/master' into patch-11

RDIL commented on tools/ci.psm1:
```diff
@@ -143,8 +143,8 @@ Function Set-BuildVariable
-        #In VSTS
-        Write-Host "##vso[task.setvariable variable=$Name;]$Value"
+        # In VSTS
+        Write-Verbose "##vso[task.setvariable variable=$Name;]$Value"
```
Oh ok

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -587,9 +594,221 @@ function Invoke-AppveyorFinish
+    }
+    else
+    {
+        Write-Verbose "Tests did not fail! Nice job!"
```
Okay

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented on tools/ci.psm1:
```diff
@@ -193,25 +193,25 @@ function Invoke-AppVeyorBuild
+    {
+        if ($env:BUILD_REASON -eq 'Schedule')
+        {
+            Write-Verbose "##vso[build.updatebuildnumber]Daily-$env:BUILD_SOURCEBRANCHNAME-$env:BUILD_SOURCEVERSION-$((get-date).ToString("yyyyMMddhhss"))"
```
Was going to do this, my git has conflicts :(

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

TravisEz13 committed:
Update tools/ci.psm1

Co-Authored-By: RDIL <jumbocakefdx@gmail.com>

RDIL commented:
Woah it got merged
Thanks 🙏
